Assets/Project/Scripts/UnityChanControlScript.cs

```diff
@@ -54,178 +63,114 @@ private void Start()
             orgColHight = col.height;
             orgVectColCenter = col.center;
 
-            // populate dictionary
-            animationStates = new Dictionary<int, Action>();
-            animationStates.Add(idleState, () => WhenIdle());
-            animationStates.Add(locoState, () => WhenRunning());
-            animationStates.Add(jumpState, () => WhenJumping());
-            animationStates.Add(restState, () => WhenResting());
-            animationStates.Add(slideState, () => WhenSliding());
-            animationStates.Add(attackState, () => WhenAttacking());
+            // Create Abilities, by default is disabled      
+            var jumpingAbility = new Ability() { AnimParamName = jumpParam, IsEnabled = false ,Key = KeyCode.Space, FuncToCall = () => JumpControls()};
+            var slidinAbility = new Ability() { AnimParamName = slideParam, IsEnabled = false ,Key = KeyCode.S, FuncToCall = () => SlidingControls()};
+            var attackAbility = new Ability() { AnimParamName = attackParam, IsEnabled = false ,Key = KeyCode.Mouse0, FuncToCall = () => AttackControls()};
+            var abilityToDie = new Ability() { AnimParamName = deathParam, IsEnabled = false, Key = KeyCode.K, FuncToCall = () => DeathControls() };
+
+            // bind ability to string name same as animation parameter
+            // careful: the enabling of the actions are performed on Abilities manager, the idea is to enable them per milestone
+            animationStates = new Dictionary<string, Ability>();
```

Comment:
Can initialise dictionary here.
`
            animationStates = new Dictionary<string, Ability>
            {
                {jumpParam, jumpingAbility},
                {slideParam, slidinAbility},
                {attackParam, attackAbility},
                {deathParam, abilityToDie}
            };
`

Reply:
yeah I find it bit clearer for other devs, when you declare it empty and then say I am adding these; But I don't mind we can initialize inside it too if preferable